### ISSUE TITLE
`DampedSinVoltage` output oscillates before `starttime`, `offset` doesn't work with negative `starttime`

### ISSUE THREAD

**Issue by AayushSabharwal** [SciML/ModelingToolkitStandardLibrary.jl#20]
`_step(t, δ, 1.0, st)` doesn't actually become 0, and instead asymptotically converges there. This works fine for stepping functions such as `_cos_wave`, but `_damped_sin_wave` oscillations grow too fast (exponentially in the reverse direction). Hence, the actual output turns into something like this (plot from Desmos)

![image](https://user-images.githubusercontent.com/23384717/145350417-0361ae3b-4570-4d91-9f07-245090f9a5d7.png)

`starttime` is set to be 10 here, so it is expected the output would be 0 before then but it isn't.

Additionally, the offset applied to the voltage of `DampedSinVoltage` doesn't respect `starttime`. The `offset` is only applied at `t>=0`, and hence if `starttime` is negative with a non-zero `offset` it leads to the following:

![image](https://user-images.githubusercontent.com/23384717/145350846-ee586ac4-f183-410e-8f2c-1a3d59186ec5.png)

This is with `offset=3.3` and `starttime=-2.8`

This is fixable by changing:
`v ~ _step(t, δ, o, 0.0) + _damped_sine_wave(t, f, A, st, ϕ, d) * _step(t, δ, 1.0, st)`
to:
`v ~ _step(t, δ, o, st) + _damped_sine_wave(t, f, A, st, ϕ, d) * _step(t, δ, 1.0, st)`

Resulting in the following:
![image](https://user-images.githubusercontent.com/23384717/145351176-e5afba5a-657c-48a2-8a92-5c38c4a22922.png)

**Comment by ven-k** [comment]
Yeah, this is an oversight. `_step` should've had `st`. I'll fix this
